ansible/roles/stack-sunbird/templates/ml-survey-service.env

```diff
@@ -0,0 +1,154 @@
+# ML Survey Service Config
+# Application host
+HOST={{ml_survey_host | default("localhost")}}
+
+# Application port number which the service will listen to
+PORT={{ml_survey_port | default("7201")}} 
+
+# Loging type
+LOG={{ml_survey_log | default("debug")}}
+
+#Application ENV example local/dev/production
+NODE_ENV={{ml_survey_node_env|default("production")}}
+
+#Log configuration
+#Enable console logging
+ENABLE_DEBUG_LOGGING={{ml_survey_enable_debug_logging | default("ON")}}
+
+# Allow report request to run for longer time
+REQUEST_TIMEOUT_FOR_REPORTS={{ml_survey_request_timeout_for_reports | default("600000")}}
+
+# User read api end point
+USER_PROFILE_FETCH_ENDPOINT={{ml_survey_user_profile_fetch_endpoint | default("/api/user/v1/read")}}
+
+## Cloud Environment platform
+# which will be templatized from the above variable (defaults)
+
+# Cloud configuration for assesment service
+# Options available
+#
+## Azure Cloud Configuration
+#
+#  CLOUD_STORAGE=AZURE
+#  AZURE_ACCOUNT_NAME=
+#  AZURE_ACCOUNT_KEY=
+#  AZURE_STORAGE_CONTAINER=
+#
+## Google Cloud Configuration
+#
+#  CLOUD_STORAGE=GCP
+#  GCP_PATH=
+#  GCP_BUCKET_NAME=
+#
+## AWS Cloud Configuration
+#
+#  CLOUD_STORAGE=AWS
+#  AWS_ACCESS_KEY_ID=
+#  AWS_SECRET_ACCESS_KEY=
+#  AWS_BUCKET_NAME=
+#  AWS_BUCKET_ENDPOINT=
+#  AWS_BUCKET_REGION=
+# To change the cloud config for azure, override below variable in common.yaml
+# For Example, 
+# ml_survey_cloud_config: |
+#   CLOUD_STORAGE=AZURE
+#   AZURE_ACCOUNT_NAME=mystorageaccount
+#   AZURE_ACCOUNT_KEY=xx123supersecretpassword
+#   AZURE_STORAGE_CONTAINER=slassessmentscontainer
+
+{{ml_survey_cloud_config}}
+
+# Mongodb connection url
+MONGODB_URL={{ml_survey_mongodb_url}}
+
+# Base url of the sunbird enviornment
+USER_SERVICE_URL={{ml_survey_user_service_URL}}
+
+# Current name for mongodb database 
+DB={{ml_survey_db_name|default("sl"+env)}}
+
+# Secure token access Admin related API's
+# The value of internal access token can be generated by command - "openssl rand -hex 10"
+INTERNAL_ACCESS_TOKEN={{ml_survey_internal_access_token}}
+
+# CSV Config
+# Create the respective directory as provided in the path below (Path to folder where reports will be stored)
+CSV_REPORTS_PATH={{ml_survey_csv_reports_path | default("reports")}}
+
+# Kafka Configuration
+# Flag which define where communication with kafka is on or off 
+KAFKA_COMMUNICATIONS_ON_OFF={{ml_survey_kafka_communications_on_off | default("ON")}}
+
+# IP address of kafka server with port without HTTP
+KAFKA_URL={{ml_survey_kafka_url}} 
+
+# Kafka topic name for pushing completed sumbmission
+COMPLETED_SUBMISSION_TOPIC={{ml_survey_completed_submission_topic | defaults (env+".sl.submission.raw")}}
+
+# Kafka topic name for pushing incompleted submission
+INCOMPLETE_SUBMISSION_TOPIC={{ml_survey_incomplete_submission_topic | defaults (env+".sl.incomplete.submission.raw")}}
```

Comment:
its not defaults, its default.
Please test the role prior.

Reply:
check for all the following variables also.

Reply:
changed